dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/JdbcEnrollmentStore.java

```diff
@@ -0,0 +1,641 @@
+      try {
+        jdbcNotes = mapper.readValue(jsonNotes, new TypeReference<>() {});
+      } catch (JsonProcessingException e) {
+        log.error("Error mapping enrollment notes: {}", jsonNotes);
+        return List.of();
+      }
+
+      List<Note> notes = new ArrayList<>();
+      for (JdbcNote jdbcNote : jdbcNotes) {
+        Note note = new Note();
+        note.setUid(jdbcNote.getUid());
+        note.setNoteText(jdbcNote.getText());
+        note.setCreator(jdbcNote.getCreator());
+        note.setCreated(formatDate(jdbcNote.getCreated()));
+        User user = new User();
+        user.setUid(jdbcNote.getUpdatedByUid());
+        user.setUsername(jdbcNote.getUpdatedByUsername());
+        user.setFirstName(jdbcNote.getUpdatedByFirstname());
+        user.setSurname(jdbcNote.getUpdatedBySurname());
+        user.setName(jdbcNote.getUpdatedByName());
+        note.setLastUpdatedBy(user);
+        notes.add(note);
+      }
+
+      return notes;
+    }
+
+    private Set<TrackedEntityAttributeValue> mapTrackedEntityAttributeValues(
+        String jsonAttributes, TrackedEntity trackedEntity) {
+      List<JdbcAttribute> attributes;
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        attributes = mapper.readValue(jsonAttributes, new TypeReference<>() {});
+      } catch (JsonProcessingException e) {
+        log.error("Error mapping enrollment attributes: {}", jsonAttributes);
+        return Set.of();
+      }
+
+      Set<TrackedEntityAttributeValue> trackedEntityAttributeValues = new HashSet<>();
+      for (JdbcAttribute attribute : attributes) {
+        TrackedEntityAttributeValue teav = new TrackedEntityAttributeValue();
+        TrackedEntityAttribute tea = new TrackedEntityAttribute();
+        tea.setUid(attribute.getUid());
+        tea.setValueType(ValueType.valueOf(attribute.getValueType()));
+        tea.setName(attribute.getName());
+        tea.setCode(attribute.getCode());
+        tea.setConfidential(attribute.isConfidential());
+        teav.setAttribute(tea);
+        teav.setStoredBy(attribute.getStoredBy());
+        teav.setCreated(formatDate(attribute.getCreated()));
+        teav.setLastUpdated(formatDate(attribute.getLastUpdated()));
+        teav.setEncryptedValue(attribute.getEncryptedValue());
+        teav.setPlainValue(attribute.getValue());
+        teav.setTrackedEntity(trackedEntity);
+
+        trackedEntityAttributeValues.add(teav);
+      }
+
+      return trackedEntityAttributeValues;
+    }
+
+    private static final ZoneId ZONE_ID = ZoneId.systemDefault();
+
+    private Date formatDate(String date) {
+      return date == null ? null : toDate(LocalDateTime.parse(date).atZone(ZONE_ID));
+    }
+
+    private Date formatDate(Timestamp timestamp) {
+      return timestamp == null ? null : toDate(timestamp.toInstant().atZone(ZONE_ID));
+    }
+
+    private Date toDate(ZonedDateTime zonedDateTime) {
+      return Date.from(zonedDateTime.toInstant());
+    }
+  }
```

Comment:
Did you check in https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/util/DateUtils.java if we don't have something already implemented here?